### PR TITLE
Fix/1319/grabbed color metric can never be released

### PR DESCRIPTION
**Pull request by JaiJoPa**
# Color Range slider buttons can now be released correctly

_**Please read the [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md) before opening a PR.**_

Fixes: #1319 

## Description

**Descriptive pull request text**, answering:
  - When pressing on a button of the the color range slider, it was never released, because stopPropagation() was called with the mouseup. That way the release was never registered.  
 - When removing the stopPropagation, the Panel is automatically closed, as we had a closing function when clicking anywhere on CodeCharta.
- That led to the problem that it also closes the Panel of color metric when clicking on the colorRange slider, as it is a child of CodeMap.
- To solve that issue,  the closing function is outsourced to codeMap, toolbar and ribbonbar. That way the Panels will only be closed when clicking on one of these components. 

## Screenshots or gifs
Panel

### CODE REVIEW

File: visualization/app/codeCharta/ui/ribbonBar/ribbonBar.component.html [hunk 1]
```diff
@@ -54,7 +52,7 @@
 <md-card
 	ng-attr-id="color-metric-card"
 	ng-class="{ expanded: $ctrl._viewModel.panelSelection === $ctrl._viewModel.panelSelectionValues.COLOR_PANEL_OPEN }"
-	ng-mouseup="$event.stopPropagation()"
+	‚
```
Comment (BridgeAR) [inline visualization/app/codeCharta/ui/ribbonBar/ribbonBar.component.html:55]:
```suggestion
```

File: visualization/app/codeCharta/ui/codeMap/codeMap.component.ts [hunk 2]
```diff
@@ -44,6 +46,10 @@ export class CodeMapController implements IsAttributeSideBarVisibleSubscriber, I
 		this.synchronizeAngularTwoWayBinding()
 	}
 
+	onClick() {
+		this.codeChartaMouseEventService.closeComponentsExceptCurrent()
```
Comment (ce-bo) [inline visualization/app/codeCharta/ui/codeMap/codeMap.component.ts:50]: Do the tests cover that closeComponentsExceptCurrent() closes all except the current one?
We might should add a second unit test for it.

**Review by ce-bo — COMMENTED** [review]
LGTM

**Review by RomanenkoVladimir — APPROVED** [review]

**Comment by RomanenkoVladimir** [comment]
Please update the Changelog

**Pull request merged**